components/builder-core/src/access_token.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2017 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::path::PathBuf;
+use chrono::Utc;
+
+use hab_net::privilege::FeatureFlags;
+use error::{Error, Result};
+use integrations::{encrypt, decrypt};
+use protocol::{message, sessionsrv};
+
+pub const BUILDER_ACCOUNT_ID: u64 = 0;
+const ACCESS_TOKEN_PREFIX: &'static str = "TOKN";
+
+const BUILDER_TOKEN_EXPIRATION_SECONDS: i64 = 7200; // 2 hours
+const USER_TOKEN_EXPIRATION_SECONDS: i64 = 315360000; // 10 years
+
+pub fn generate_bldr_token(key_dir: &PathBuf) -> Result<String> {
+    generate_access_token(
+        key_dir,
+        BUILDER_ACCOUNT_ID,
+        FeatureFlags::all().bits(),
+        BUILDER_TOKEN_EXPIRATION_SECONDS,
+    )
+}
+
+pub fn generate_user_token(key_dir: &PathBuf, account_id: u64, privileges: u32) -> Result<String> {
+    generate_access_token(
+        key_dir,
+        account_id,
+        privileges,
+        USER_TOKEN_EXPIRATION_SECONDS,
+    )
+}
+
+pub fn generate_access_token(
+    key_dir: &PathBuf,
+    account_id: u64,
+    flags: u32,
+    expiration: i64,
+) -> Result<String> {
+    let expires = Utc::now().timestamp() + expiration;
+
+    let mut token = sessionsrv::AccessToken::new();
+    token.set_account_id(account_id);
+    token.set_flags(flags);
+    token.set_expires(expires);
+
+    let bytes = message::encode(&token).map_err(Error::Protocol)?;
+    let ciphertext = encrypt(key_dir, &bytes)?;
+
+    Ok(format!("{}{}", ACCESS_TOKEN_PREFIX, ciphertext))
+}
+
+pub fn is_access_token(token: &str) -> bool {
+    token.starts_with(ACCESS_TOKEN_PREFIX)
+}
+
+pub fn validate_access_token(key_dir: &PathBuf, token: &str) -> Result<sessionsrv::Session> {
+    if !token.starts_with(ACCESS_TOKEN_PREFIX) {
```

Comment:
Might just use `is_access_token(token)` here instead.

Reply:
Yes!  this used to be assert using is_access_token, at one point I had removed that function and forgot to put it back as it was - good catch.